app/pages/planner/page.tsx

```diff
@@ -93,36 +93,21 @@ export default function Planner() {
   }, []);
 
 
-  const onSearchClick = (searchValue: string) => {
-    fetch(
-      "/api/courses?" +
-        new URLSearchParams({
-          textSearch: searchValue,
-          size: "10"
-        }),
-      {
-        method: "GET",
-        headers: {
-          "Content-Type": "application/json",
-        },
+  const onSearchClick = (searchValue: string, courseList: CourseDTO[]) => {
+    // Filter out courses that are already selected
+    const res = courseList.filter((c) =>
+      c.course_code.toLowerCase().includes(searchValue.toLowerCase())
+    );
+    let outputData = [];
+    for (const element of res) {
+      if (
+        selected.course.filter((c: CourseDTO) => c.id === element.id)
```

Comment:
Would it be more conventional to use `!selected.course.filter((c: CourseDTO) => c.id === element.id)`?